src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateController.php

```diff
@@ -196,12 +195,10 @@ public function copyAction(Request $request, $manageId, $targetEnvironment, $sou
         if (!$request->isMethod('post')) {
             $entityId = $this->entityService->createEntityUuid();
 
-            $this->commandBus->handle(
-                new CopyEntityCommand($entityId, $manageId, $service, $targetEnvironment, $sourceEnvironment)
-            );
+            // copy entity
+            $entity = $this->copyEntityService->copy($entityId, $manageId, $service, $targetEnvironment, $sourceEnvironment);
```

Comment:
Why break with our CQRS principle here? Unless you have a very good reason to do this, I think it should be reverted to the previous way. Maybe you can keep the newly created service and use it in the CommandHandler you reintroduce. This way we have best of both worlds!

Reply:
The entity copy is on the read side of CQRS because it currently doesn't write anything but only reads data to inject into the form. I agree that it's not very intuitive and therefore I renamed the service to LoadEntityService to reflect better what it actually does.